pkg/processor/transform/txfbuiltin/httprequest.go

```diff
@@ -0,0 +1,106 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package txfbuiltin
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/processor"
+	"github.com/conduitio/conduit/pkg/processor/transform"
+	"github.com/conduitio/conduit/pkg/record"
+)
+
+const (
+	httpRequestName = "httprequest"
+
+	httpRequestConfigURL    = "url"
+	httpRequestConfigMethod = "method"
+)
+
+func init() {
+	processor.GlobalBuilderRegistry.MustRegister(httpRequestName, transform.NewBuilder(HTTPRequest))
+}
+
+// HTTPRequest builds a transform that sends an HTTP request to the specified
+// URL with the specified HTTP method (default is POST). The record payload is
+// used as the request body and the raw response body is put into the record
+// payload.
```

Comment:
Since headers are left out (maybe out of scope for now?), we will likely not be able to authenticate. Also, some endpoints will require certain headers to be present (e.g. Content-Type), or the request cannot be processed.

Reply:
Minimal is ok, though, it will be nice to have some rudimentary retry policy that could handle blips. Timeouts as well.

Reply:
I implemented a configurable backoff retry that is disabled by default, but can be enabled by setting `backoffRetry.count` to a number greater than 0. You can also fine-tune the backoff retry with `backoffRetry.min`, `backoffRetry.max` and `backoffRetry.factor` (the delay between retries starts with min and is multiplied by factor until it reaches max). Any request that returns a status code greater than 299 will be retried.

@lyuboxa let me know if that's what you had in mind.

Reply:
looks good

Reply:
Yes, we can do for now ;-)